### PR TITLE
Chapter 10 Exercise 04 complete

### PR DESCRIPTION
**Pull request by archiebrowne**
with documentation + status update

### CODE REVIEW

File: M1FExplained/Chapter10/Exercises/Exercise03.lean [hunk 1]
```diff
@@ -68,7 +67,9 @@ at any even hour on some day. Similarly if a train leaves at 11, it is possible
 at any odd hour.
 -/
 
-lemma part_c_i (t y : ℕ) (h : t < 24 ∧ y < 24) (hpara : Even t ∧ Even y) : ∃ (m : ℕ), t + 14 * m ≡ y [MOD 24] := by sorry
+lemma part_c_i (t y : ℕ) (h : t < 24 ∧ y < 24) (hpara : Even t ∧ Even y) : ∃ (m : ℕ), t + 14 * m ≡ y [MOD 24] := by 
+  
+sorry
```
Comment (kbuzzard) [inline M1FExplained/Chapter10/Exercises/Exercise03.lean]:
```suggestion
lemma part_c_i (t y : ℕ) (h : t < 24 ∧ y < 24) (hpara : Even t ∧ Even y) : 
    ∃ (m : ℕ), t + 14 * m ≡ y [MOD 24] := by 
  sorry
```

File: M1FExplained/Chapter10/Exercises/Exercise04.lean [hunk 2]
```diff
@@ -2,48 +2,77 @@ import Mathlib.Tactic
...
+state just one of these assumptions and deduce the other based on the fact that `a ∣ b` and `b ∣ a`
+with the following (choosing `a ≠ 0` WLOG):
+-/
 
-  -/
+example (a b : ℤ) (ha : a ≠ 0) (hab : a ∣ b) (hba : b ∣ a) : b ≠ 0 := by
+  intro h
+  rw [h] at hba
+  simp at hba
+  contradiction
```
Comment (kbuzzard) [inline M1FExplained/Chapter10/Exercises/Exercise04.lean]:
```suggestion
  simp_all
```

File: M1FExplained/Chapter10/Exercises/Exercise04.lean [hunk 3]
```diff
@@ -2,48 +2,77 @@ import Mathlib.Tactic
...
   match hab with
   |⟨k₁, hk₁⟩ => 
   match hba with
   |⟨k₂, hk₂⟩ => 
   rw [hk₂] at hk₁
+  -- If `b = a * k₁` and `a = b * k₂`, then it must be that `k₁ * k₂  = 1`.
   have h₁ : 1 = k₂ * k₁ := by 
     calc
-      1 = b / b := sorry --exact (Eq.symm (Int.div_self hzero.right))
+      1 = b / b := by exact Eq.symm (Int.ediv_self hb)
```
Comment (kbuzzard) [inline M1FExplained/Chapter10/Exercises/Exercise04.lean]:
```suggestion
      1 = b / b := by rw [Int.ediv_self hb]
```
Or just cancel the by and exact and leave the Eq.symm in.

**Review by kbuzzard — COMMENTED** [review]
Thanks! I've left some comments.

**Comment by archiebrowne** [comment]
Thanks, committed the changes

**Comment by kbuzzard** [comment]
Can you fix the conflict and then this looks good to go.

**Comment by kbuzzard** [comment]
Thanks!

**Pull request merged**